### DISCUSSION TITLE
Delete some Z_ scripts

### DISCUSSION THREAD

**Discussion by pakricard** [timschofield/webERP#281] [Development Discussion & Specification]
Hi all:

Some Z_ scripts are quite dangerous, and even some have the comment "STRONGLY RECOMMEND NOT USING THIS "
e.g.: Z_DeleteInvoice.php, Z_DeleteCreditNote.php

Some are most probably obsolete and inconsistent with current DB as the last substantial commit was done long ago (strong indicator no one is using them).

- Z_CreateCompany.php, last substantial commit was done 2005.
- Z_CreateCompanyTemplateFile.php last substantial commit was done 2013.
- Z_CurrencyDebtorsBalances.php last substantial commit was done 2011.
- Z_CurrencySuppliersBalances.php last substantial commit was done 2011.
- Z_DataExport.php last substantial commit was done 2010.

Should we get rid of these scripts because some are dangerous or most probably inconsistent for more than 10 years?

**Comment by dalers** [comment]
Hi @pakricard

Do menu items exist for any of these scripts? 

Presumably all were created to solve some problem, but perhaps the problems were unique situations and possibly even only temporary.

How about moving them into docs/scripts with a README saying they are for guidance only, not likely to run without changes, and should not be run on a live database without testing on a test database first. Moving the scripts would cleanup the main directory and someone could still benefit (and then delete entirely in another year if truly have no value).

**Comment by pakricard** [comment]
Most of them do have a menu entry (at least in v 4.15.2). 

We seldomly use:
- Z_Change_XXX as sometimes we need to update a location code, stockid, etc. 
- Z_RePostGLFromPeriod and Z_UpdateChartDetailsBFwd are safe to use and used when something goes off on the GL side with chartdetails
- Z_ItemsWithoutPicture is safe to use
These are kept up-to-date and not risky to use.

I am not sure why the name starts with "Z_" Being safe scripts we could probably use standard script names.

All Z_Import files are most likely obsolete. I would not use them in my webERP without a meticulous check as can lead to inconsistencies,

**Reply by dalers** [reply]
> _**Most**_ of them do have a menu entry (at least in v 4.15.2).

Can anyone suggest a general way (i.e. bash script?) to determine _**exactly**_ a) which php files have menu items, and b) which files are **not** included in or executed from other php files?

I'm guessing all the files listed in PageSecurity.php would likely have assigned menu items, but I suspect there's nothing in place to require it (there could be files listed in PageSecurity.php that aren't in the menu, and there could maybe be php files in the menu that are not listed). @timschofield is this possible?

**Comment by timschofield** [comment]
I think most of them can go. Many of them were done as a fix for problems
users were having - for instance at one time we were constantly asking
people to run Z_RePostGLFromPeriod.php when their TB went out of balance.
This is now redundant as I cured the underlying problem. Z_DeleteInvoice
and Z_DeleteCreditNote come from a time when users kept asking for this
functionality, and Phil and I were constantly saying it was bad accounting
practice. In the end somebody wrote them (not me) and they were given the
caveat that we don't support them.
A few of them are useful (such as uploading initial transactions).

People would be more likely to maintain these scripts if there were fewer
of them, so I would definitely encourage heavily pruning them. Maybe a
start would be to remove the utilities module altogether? The scripts would
still be available but users would either have to be directed there, or
know to look.

Dose anyone have any particular favourites they would like to keep?


On Fri, 6 Dec 2024 at 23:25, Dale Scott ***@***.***> wrote:

> *Most* of them do have a menu entry (at least in v 4.15.2).
>
> Can anyone suggest a general way (i.e. bash script?) to determine a) which
> php files have menu items, and b) which files are not included in or
> executed from other php files?
>
> I'm guessing all the files listed in PageSecurity.php would likely have
> assigned menu items, but I suspect there's nothing in place to require it
> (there could be files listed in PageSecurity.php that aren't in the menu,
> and there could maybe be php files in the menu that are not listed).
> @timschofield <https://github.com/timschofield> is this possible?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/281#discussioncomment-11490741>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LFSARYUMVGXGOA2BQD2EIW5JAVCNFSM6AAAAABTD3BFGKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBZGA3TIMI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by pakricard** [comment]
My vote is to keep the Utilities menu, as it can be easily enabled/disabled via WWW_Users.php). Some of the scripts are "admin only" or seldomly used but practical, like the Z_Change* ones.

Furthermore, deleting dangerous scripts should be a priority as their usage is not worth the risk. IMO, these will not be missed:
- Z_DeleteInvoice
- Z_DeleteCreditNote 
- Z_CreateCompany.php
- Z_CreateCompanyTemplateFile.php
- Z_CurrencyDebtorsBalances.php 
- Z_CurrencySuppliersBalances.php 
- Z_DataExport.php

**Comment by timschofield** [comment]
Hi Ricard, my two pennorth below:

I am ambivalent about this one.

On Mon, 9 Dec 2024 at 02:00, pakricard ***@***.***> wrote:
>
> My vote is to keep the Utilities menu, as it can be easily enables/disabled via WWW_Users.php). Some of the scripts are "admin only" or seldomly used but practical, like the Z_Change* ones.

The use of the utilities is relatively recent - I have a feeling I did
it originally. Before then the Z_ scripts existed, but users were
guided to them from the mailing lists.
The Z_Change scripts I can live with, but the accountant in me is
always nervous about such things.

>
> Furthermore, deleting dangerous scripts should be a priority as their usage is not worth the risk. IMO, these will not be missed:
>
> Z_DeleteInvoice
> Z_DeleteCreditNote

Agreed with these 2, however they were introduced because users kept
asking for them. Basically it was a case of "QuickBooks lets you do it
why doesn't webERP?" I wouldn't have done them personally but they are
there.

> Z_CreateCompany.php

I think this script, if implemented right is a good one. Imagine for
instance someone installs the demo data, plays around with it and
eventually decides to install it for their organisation. It would be
much neater for them to be able to create a new blank company for
production purposes, but keeping the demo for practising/training. I
know this can be done manually but it's so much nicer using the
script. Having said that the script needs work.

> Z_CreateCompanyTemplateFile.php

I think this is obsoleted by the one above

> Z_CurrencyDebtorsBalances.php
> Z_CurrencySuppliersBalances.php

These two don't update anything, though I don't really see their
point. Why have you classed them as "dangerous"?

> Z_DataExport.php

I think I agree here as well. Unless a developer is prepared to
regularly maintain this script it rapidly breaks. Someone who knows
what they are doing can easily export the DB if they know what they
are doing.

>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

Thanks
Tim


-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by timschofield** [comment]
I have created a wiki page -
https://github.com/timschofield/webERP/wiki/Utilities - where people
can put comments against scripts the pafrticularly want to
keep/scratch

On Mon, 9 Dec 2024 at 11:54, Tim Schofield ***@***.***> wrote:
>
> Hi Ricard, my two pennorth below:
>
> I am ambivalent about this one.
>
> On Mon, 9 Dec 2024 at 02:00, pakricard ***@***.***> wrote:
> >
> > My vote is to keep the Utilities menu, as it can be easily enables/disabled via WWW_Users.php). Some of the scripts are "admin only" or seldomly used but practical, like the Z_Change* ones.
>
> The use of the utilities is relatively recent - I have a feeling I did
> it originally. Before then the Z_ scripts existed, but users were
> guided to them from the mailing lists.
> The Z_Change scripts I can live with, but the accountant in me is
> always nervous about such things.
>
> >
> > Furthermore, deleting dangerous scripts should be a priority as their usage is not worth the risk. IMO, these will not be missed:
> >
> > Z_DeleteInvoice
> > Z_DeleteCreditNote
>
> Agreed with these 2, however they were introduced because users kept
> asking for them. Basically it was a case of "QuickBooks lets you do it
> why doesn't webERP?" I wouldn't have done them personally but they are
> there.
>
> > Z_CreateCompany.php
>
> I think this script, if implemented right is a good one. Imagine for
> instance someone installs the demo data, plays around with it and
> eventually decides to install it for their organisation. It would be
> much neater for them to be able to create a new blank company for
> production purposes, but keeping the demo for practising/training. I
> know this can be done manually but it's so much nicer using the
> script. Having said that the script needs work.
>
> > Z_CreateCompanyTemplateFile.php
>
> I think this is obsoleted by the one above
>
> > Z_CurrencyDebtorsBalances.php
> > Z_CurrencySuppliersBalances.php
>
> These two don't update anything, though I don't really see their
> point. Why have you classed them as "dangerous"?
>
> > Z_DataExport.php
>
> I think I agree here as well. Unless a developer is prepared to
> regularly maintain this script it rapidly breaks. Someone who knows
> what they are doing can easily export the DB if they know what they
> are doing.
>
> >
> > —
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> > You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> Thanks
> Tim
>
>
> --
> www.weberp.org
> @TimSchofield2
> Blog: https://kwamoja.home.blog/



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by pakricard** [comment]
Hi @timschofield :  

Z_CurrencyDebtorsBalances.php and  Z_CurrencySuppliersBalances.php are safe to run (confusing long sentence). Just I don't think they belong to Utilities. If they are useful to someone, they should be somewhere in Receivables or Payables menu.
Z_ItemsWithoutPicture should be somewhere in the inventory menu.

`The Z_Change scripts I can live with, but the accountant in me is always nervous about such things.`

Z_Change are mainly used against undetected typos in codes, or when standard coding changes, not on regular basis.

All the Z_Update used to change DB versions are very old. Maybe someone is still using v 3.01 and wants to upgrade to v5 but... I really doubt it. Some old versions need to be discontinued at one point. 

To me, this is an exercise of mental sanity. The less old, unused, potentially dangerous scripts we have to maintain, the better. In fact, in my install I deleted almost all of them.

**Reply by dalers** [reply]
It seems there are a number of categories of "Z" scripts:

- scripts with menu entries in Utilities that would be "better" if their menu entries were in an "operational module" menu e.g. Sales, Receivables, Purchases, Inventory... because it would make them more intuitive to find.
- scripts that perform maintenance functions (e.g. report on missing images, delete an invoice or credit note created in error...)
- scripts of a temporary nature used to correct database issues caused by code issues that were later resolved
- scripts that are redundant (superceded by another script that provides the same or improved functionality)
- scripts that fail or work incorrectly but still have benefit as guidance to developers

If the scripts can be suitably categorized, we could decide on an appropriate action for each category.

For example, we could decide:

- scripts with menu entries in Utilities that would be "better" in a "module" menu are renamed removing the "Z_" prefix and given menu entries in the appropriate module
- scripts that provide often-required maintenance functions are renamed removing the "Z_" prefix and given menu entries in a new "Maintenance" module
- scripts of a temporary nature are moved to maintenance/
- scripts that do not function correctly or are redundant, but still useful for guidance, are moved to docs_scripts/ (otherwise deleted)

I would generalize @pakricard recommendation as:
> The less ... scripts we have to maintain, the better. 

Any thoughts for how to progress from discussion to action?

Fwiw, a couple additional related thoughts (although they infringe on the general directory structure discussion in [Redefining the directory structure (??) #189](https://github.com/timschofield/webERP/discussions/189):

- scripts that do not have a menu entry but are invoked from a script that does are in the includes/ directory
- "non-Z_" scripts that do not have menu entries or are not invoked from scripts with menu entries are renamed adding "Z_" prefix (e.g. rename LoggedInUsers.php to Z_LoggedInUsers.php)
- provide a directory reserved for users to add their own scripts ("custom" scripts) to simplify maintenance of a site with custom scripts by not mixing them with "core" scripts.

**Comment by timschofield** [comment]
Hi Dale,

On Tue, 17 Dec 2024 at 07:05, Dale Scott ***@***.***> wrote:
>
>
> I would generalize @pakricard recommendation as:
>
> The less ... scripts we have to maintain, the better.

Agreed, I think most developers (including me) forget to ask
themselves "Does this impact on a Z_ script?" meaning they quickly get
outdated.

>
> Any thoughts for how to progress from discussion to action?
>
> Fwiw, a couple additional related thoughts (although they infringe on the general directory structure discussion in Redefining the directory structure (??) #189:

I think the flat directory structure is actually a *featiure* of
webERP. It has been like it from day 1, and I wouldn't want to play
with that. The Z_ prefix was devised to group utility scripts together
in one place without the need for a separate directory.

>
> scripts that do not have a menu entry but are invoked from a script that does are in the includes/ directory

Agree that scripts that are only included from others should be in the
includes/ directory. Do you have any in mind that aren't? However some
scripts are actually invoked from other scripts (from
SelectProduct.php, SelectCustomer.php, SelectSupplier.php for
instance) and these should remain in the main directory.
AddCustomerNotes.php and AddCustomerTypeNotes.php spring to mind as
examples of this.

> "non-Z_" scripts that do not have menu entries or are not invoked from scripts with menu entries are renamed adding "Z_" prefix (e.g. rename LoggedInUsers.php to Z_LoggedInUsers.php)

LoggedInUsers.php does have a menu item for it and we are trying to
cut down on the Z_ scripts so I am not sure why we would add to them
unnecessarily.

> provide a directory reserved for users to add their own scripts ("custom" scripts) to simplify maintenance of a site with custom scripts by not mixing them with "core" scripts.

In my experience most customisations are adaptations of existing
scripts rather than new scripts. Obviously we can't control where
users put custom scripts but I would encourage them to keep their
scripts in the main directory.

Thanks
Tim

-- 
www.weberp.org

**Reply by dalers** [reply]
> I think the flat directory structure is actually a *feature* of webERP.
 
Agree, it's a feature of webERP that it is as close to bare metal (a businesses financial position) as you can get. Only enough structure to improve clarity and avoid absolute chaos...

> The Z_ prefix was devised to group utility scripts together in one place without the need for a subdirectory

Agree there is convenience. It seems "Z_" means Utilities if all scripts have a menu item. Is that right?

Why not carry the strategy further and name all scripts with the prefix of their primary module (SA_ = Sales, MF_ = Manufacturing, PY_ = Payables, PC_ = Petty Cash...)?

> Agree that scripts that are only included from others should be in the includes/ directory. Do you have any in mind that aren't? 

Don't know if there actually are any but it's good to have a rule to apply if any are found.

I've been noodling on how to run a report to find out whether all the files in the webroot do have menu entries and which scripts invoke scripts in includes/. I.e. something that would report deterministically, that can't be argued with, that can be run until it shows everying is "fixed" and could be run periodically to show fixed things are still fixed.

Any ideas? I seem to recall NetBeans can analyze a PHP project and create a call tree / graph...

> some scripts are ... invoked from other scripts... and these should remain in the main directory.

(e.g. AddCustomerNotes.php and AddCustomerTypeNotes.php are invoked from SelectProduct.php, SelectCustomer.php and SelectSupplier.php, iiuc)

What is a rule we could decide on for files in the main directory? Can a rule be made for when exceptions are allowed?

> LoggedInUsers.php does have a menu item

Sorry, didn't realize it had a menu item. Ahh, you put it in Setup, I had expected Utilities.

> we can't control where users put custom scripts but I would encourage them to keep their scripts in the main directory.

Perhaps recommending a XX_ prefix?

**Comment by timschofield** [comment]
Hi Dale,

On Tue, 17 Dec 2024 at 18:47, Dale Scott ***@***.***> wrote:
> Why not carry the strategy further and name all scripts with the prefix of their primary module (SA_ = Sales, MF_ = Manufacturing, PY_ = Payables, PC_ = Petty Cash...)?

Some of them already are - PC and GL for instance. We should probably
have done it from the start, though with some scripts the line is
blurred - AP and Purchasing, or AR and Sales. I'm not sure we should
do a mass change like that now though.

>
> LoggedInUsers.php does have a menu item
>
> Sorry, didn't realize it had a menu item. Ahh, you put it in Setup, I had expected Utilities.

The setup module is misnamed really. It started out being for
parameter setup. It grew over the years to include anything used in
the admin of the project - which is what LoggedInUsers.php is.

Utilities is for scripts that should only be used by an expert, and
could well do damage if used incorrectly.

Tim



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by pakricard** [comment]
My 2 cents:

> I think the flat directory structure is actually a feature of webERP.

I don't see an issue with that. I am not sure if a tree-like script structure helps us in any significant way.

> Some of them already are - PC and GL for instance. We should probably
have done it from the start, though with some scripts the line is
blurred - AP and Purchasing, or AR and Sales. I'm not sure we should
do a mass change like that now though.

100% agree

> In my experience most customisations are adaptations of existing
> scripts rather than new scripts. Obviously we can't control where
> users put custom scripts but I would encourage them to keep their
> scripts in the main directory.

I have both. When a small change is needed, it is easier to keep the standard webERP script and do some mods here and there, while when a new functionality is needed, new scripts are coded and kept in the main directory. To locate them easily, I added a prefix “KL” as KLScriptName.php. Good enough for me.

 
> Utilities is for scripts that should only be used by an expert, and
> could well do damage if used incorrectly.

That was my point when I started the discussion. If a script could “do damage”, it should be fixed or deleted.  My vote is for deletion.

IMO: the current code is still not PHP8 ready, has several bugs and we are a small team. I think that we should focus on getting what we have now working properly and, at a later stage, apply all improvements we might think of.

Should we go (for now) for an "if it works, don't touch it" approach until v5?

**Comment by timschofield** [comment]
On Tue, 17 Dec 2024 at 23:41, pakricard ***@***.***> wrote:
>
>
> Should we go (for now) for an "if it works, don't touch it" approach until v5?
>
> —

Agreed



--
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by pakricard** [comment]
Even if I said, _"if it works, don't touch it" approach until v5_ I think there is a Z_ script that deserves special attention: Z_UpgradeDatabase.php.

Z_UpgradeDatabase.php does now all the heavy lifting of upgrading webERP DB from version 4.15.2 onwards. Then, we have the old UpgradeDatabase.php that upgrades from 3.00 until 4.15.2

Z_UpgradeDatabase.php is now a mature script, and probably it deserves to get rid of the Z_ stigma, and we should rename it to UpgradeDatabase.php or similar and delete the previous UpgradeDatabase.php

If a user is coming from a version before 4.15.2 could be asked to upgrade to 4.15.2 first, and as a second step upgrade to the current version we are on.

Moreover, it would allow us to:

- remove all old upgradeXXX-YYY.sql files from v5 onwards. 
- remove bits and bobs of code that cater for old versions before 4.15.2 (specially in session-like scripts)
 
Fewer lines of code to maintain, cleaner code, easier life. ;-)

R

**Comment by timschofield** [comment]
In principle you are right, I would love to remove all the old upgrade stuff. The trouble is I often get emails from organisations running very old versions asking how to upgrade to a later version. Last year I got an email from someone running 3.11!!!! I can't begin to imagine when that was released but it must be 15+ years ago.

Z_CreateChartDetails.php can definitely go as the infamous chartdetails table has been consigned to history!

Tim

**Comment by dalers** [comment]
going back to an earlier comment from you @pakricard 

> All Z_Import files are most likely obsolete. I would not use them in my webERP....

Can you clarify? I saw you made editing commits to both Z_ImportStocks.php and Z_ImportPartcodes.php in 2024/2025 so I thought you were using them. Fwiw I see they were both first committed by TIm in Sept/Dec 2008 but only Z_ImportStocks.php seems to have a menu entry although their commits imply both were used.

Why is Z_ImportStocks.php linked in the menu and Z_ImportPartCodes is not? Is there a use-case for both? Iiuc the script linked in the menu (ImportStocks.php) also requires a location code in addition to stockmaster values (and also writes to the lockstock table). Are they otherwise equivalent? Do we need both?

Open question - when migrating to webERP, what is your best way to import existing part, bom and supplier data??

**Reply by timschofield** [reply]
If I remember correctly (and it was 17 years ago) I wrote Z_ImportPartcodes.php in response to a request for an example of how to use the api. I think Z_ImportStocks.php was a more serious attempt to write an import mechanism.

It's important to be able to import stock items as part of an implementation. I have implemented systems in organisations that have tens of thousands of stock items.

Tim

**Reply by dalers** [reply]
Thanks @timschofield that explains all the error checking in Z_ImportStocks.php compared to Z_ImportPartCodes.php. It would be interesting to compare the import speed of the two.

> tens of thousands of stock item

Not surprising. Fwiw, metrics from a couple companies I'm worked with:

1. an SME industrial control device manufacturer with one core product, 6 product families and a potential of 1.25M sales PNs due to length and processs connection options: 1K PNs (not including sales PNs, created if needed when ordered), 70 BOMs, 50 Suppliers and 125 OEMs (the OEM is specified although parts are purchased from Suppliers).

2. A smart-home product design and manufacturing startup with one core product and 8 variants: 350 PNs, 60 BOMs, 24 Suppliers, 70 OEMs. 

3. An industrial test services provider with 3 core test instruments and custom variants: 30K PNs, 5K BOMs, 160 Suppliers, 500 OEMs.

> It's important to be able to import stock items as part of an implementation.

Agree 150% but how have you typically done that? Using Z_ImportStocks.php, Z_ImportPartCodes.php, or a custom script for the occassion (possibly based on one of the two).

**Reply by pakricard** [reply]
> going back to an earlier comment from you @pakricard
> 
> > All Z_Import files are most likely obsolete. I would not use them in my webERP....
> 
> Can you clarify? I saw you made editing commits to both Z_ImportStocks.php and Z_ImportPartcodes.php in 2024/2025 so I thought you were using them. Fwiw I see they were both first committed by TIm in Sept/Dec 2008 but only Z_ImportStocks.php seems to have a menu entry although their commits imply both were used.

Hi @dalers : A vast majority of recent (2024-2025) commits are "general commits" a.k.a. syntax cleaning, capitalization, etc. improvements done to all scripts in webERP.

I meant "most likely obsolete" because I can't remember any functionality commit to these files. Probably my memory is failing me, ;-) but we have been adding fields here and there during the years, and I'm not sure the Z_import (and perhaps too, the API functions) have been kept up-to-date. As example Z_ImporStocks last functional commit was in 2018, commit e0daaed8a9859f41f50821a3339de627b277d954 . 

Maybe I am wrong and the Z_Import*.php are up-to-date, but before using them, I would revise in full detail.

R

**Reply by dalers** [reply]
Thanks for clarifying @pakricard, I think I understand better now. I absolutely think Z_ImportStocks.php needs to stay because It's linked in the menu and presumably is (or was meant to be) the officially supported method to import stock codes. It may be that I thought some fields are optional or have default values when they are actually required. I'll open an issue if I can't figure it out myself.

However, I suggest Z_ImportPartCodes.php be renamed "api_ImportPartCodes.php" and moved to a new ./doc/examples/ directory as an example of how to use the API, and move all the other files that can't be reached from the menu system into examples/ as well.

**Reply by timschofield** [reply]
Having said all this, Z_ImportPartCodes.php is probably a better model for how to do these import scripts as it doesn't need updating when the table changes.

Tim

**Reply by dalers** [reply]
@timschofield, is the error correction that is done in ImportStocks.php done in the API instead when ImportPartCodes.php is used? (easier to ask than look....  ;-))

- (edited after looking) Yes, the API performs what looks to be the same error correction.

Can you describe a class of scripts (e.g., all the import scripts?) that should use the API instead of accessing the database directly?

- (edited) @timschofield already answered in part in his previous comment by saying "Z_ImportPartCodes.php is ... a better model for how to do these import scripts". I.e. at least all import scripts should use the API.

**Reply by dalers** [reply]
> Having said all this, Z_ImportPartCodes.php is probably a better model for how to do these import scripts

@timschofield does that mean you would be in favor of [Main > Utilities > Maintenance > Import Stock Items from .csv] running Z_ImportPartCodes.php instead of Z_ImportStocks.php?

**Reply by timschofield** [reply]
In theory, yes, but it would need proper testing first

**Comment by dalers** [comment]
Still poking around importing and noticed Z_DescribeTable.php is used by Z_ImportPartCodes.php and also that it can't be used as a stand-alone script (because it needs a table name).

Because Z_DescribeTable.php can't be used stand-alone, would it be more "correct" wrt the project file structure if it was moved to the ./includes/ directory?

**Reply by timschofield** [reply]
Sounds like a good idea to me Dale

**Reply by dalers** [reply]
New issue #729 has been created to deal with Z_DescribeTable.php.
